app/Services/MaintenanceWindowService.php

```
@@ -0,0 +1,368 @@
+     */
+    protected function getWindowEnd(array $window, Carbon $start): Carbon
+    {
+        $type = $window['type'] ?? null;
+
+        if ($type === 'one_time') {
+            return Carbon::parse($window['end']);
+        }
+
+        if ($type === 'recurring') {
+            $endTime = $window['end_time'] ?? '00:00';
+            $timezone = $window['timezone'] ?? config('app.timezone', 'UTC');
```

PHPMD (2.15.0)

Avoid unused local variables such as '$timezone'. (undefined)

(UnusedLocalVariable)